games/xworld/maps/XWorldDialogMap.py

```diff
@@ -12,8 +12,8 @@ def __init__(self, item_path, start_level=0):
 
     def _configure(self, select_class=True):
         self.set_dims(1, 1)
-        self.set_entity(type="agent")
-        self.set_entity(type="goal")
+        self.set_entity(type="agent", loc=(0, 0), force_occupy=True)
+        self.set_entity(type="goal", loc=(0, 0), force_occupy=True)
         if select_class:
             self.select_goal_classes() # re-select goal class for a new session
```

Comment:
this will have conflicts if later a maze is randomly generated.

Reply:
Now makes the maze_generation and optional choice and generates warning message for over-writing pre-specified locations if the maze generation is turned on.